contracts/products/SoteriaCoverageProduct.sol

```diff
@@ -0,0 +1,681 @@
+    }
+
+    /**
+     * @notice set _maxRateDenom.
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @param maxRateDenom_ Desired maxRateDenom.
+    */
+    function setMaxRateDenom(uint256 maxRateDenom_) external override onlyGovernance {
+        _maxRateDenom = maxRateDenom_;
+        emit MaxRateDenomSet(maxRateDenom_);
+    }
+
+    /**
+     * @notice set _chargeCycle.
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @param chargeCycle_ Desired chargeCycle.
+    */
+    function setChargeCycle(uint256 chargeCycle_) external override onlyGovernance {
+        _chargeCycle = chargeCycle_;
+        emit ChargeCycleSet(chargeCycle_);
+    }
+
+    /**
+     * @notice set _referralRewardPercentage
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @param referralRewardPercentage_ Desired referralRewardPercentage.
+    */
+    function setReferralRewardPercentage(uint256 referralRewardPercentage_) external override onlyGovernance {
+        require(referralRewardPercentage_ <= 10000, "cannot set over 100%");
+        _referralRewardPercentage = referralRewardPercentage_;
+        emit ReferralRewardPercentageSet(referralRewardPercentage_);
+    }
+
+    /***************************************
+    COVER PROMOTION ADMIN FUNCTIONS
+    ***************************************/
+
+    /**
+     * @notice Enables cover promotion admin to gift (and remove) 'free' cover to specific addresses.
+     * Can only be called by the current cover promotion admin.
+     * @param policyholder_ The policy holder to set reward points for.
+     * @param rewardPoints_ Desired amount of reward points.
+    */
+    function setRewardPoints(address policyholder_, uint256 rewardPoints_) external override {
+        require(msg.sender == _registry.get("coverPromotionAdmin"), "not cover promotion admin");
+        _rewardPointsOf[policyholder_] = rewardPoints_;
+        emit RewardPointsSet(policyholder_, rewardPoints_);
+    }  
+
+    /***************************************
+    PREMIUM COLLECTOR FUNCTIONS
+    ***************************************/
+
+    /**
+     * @notice Charge premiums for each policy holder.
+     * @param holders_ The policy holders.
+     * @param premiums_ The premium amounts in `wei` per policy holder.
+    */
+    function chargePremiums(address[] calldata holders_, uint256[] calldata premiums_) external payable override whileUnpaused {
+        uint256 count = holders_.length;
+        require(msg.sender == _registry.get("premiumCollector"), "not premium collector");
+        require(count == premiums_.length, "length mismatch");
+        require(count <= policyCount(), "policy count exceeded");
+        uint256 amountToPayPremiumPool = 0;
+
+        for (uint256 i = 0; i < count; i++) {
+            // skip computation if policy inactive
+            if ( !policyStatus(_policyOf[holders_[i]]) ) continue;
```

Comment:
If a user deactivated policy premiumCollector must still be able to charge a premium (e.g., last charge was on Monday, user calls deactivate on Wed, premiumCollector need to charge 2 days…)

Reply:
I think the solution to this is to use the  _preDeactivateCoverLimitOf mapping you commented on here - https://github.com/solace-fi/solace-core/pull/246#discussion_r792284993

Reply:
sounds good, that could work, please push the fix